pkg/bazel/bazel.go

```diff
@@ -156,20 +145,40 @@ func (b *bazel) Flags() (map[string]*flags.FlagInfo, error) {
 		return nil, fmt.Errorf("failed to get Bazel flags: %w", err)
 	}
 
-	flags := make(map[string]*flags.FlagInfo)
+	availableStartupFlags := []string{}
 	for i := range flagCollection.FlagInfos {
-		flags[*flagCollection.FlagInfos[i].Name] = flagCollection.FlagInfos[i]
-		for _, command := range flags[*flagCollection.FlagInfos[i].Name].Commands {
+		allFlags[*flagCollection.FlagInfos[i].Name] = flagCollection.FlagInfos[i]
+		for _, command := range allFlags[*flagCollection.FlagInfos[i].Name].Commands {
 			if command == "startup" {
 				availableStartupFlags = append(availableStartupFlags, *flagCollection.FlagInfos[i].Name)
-				if flags[*flagCollection.FlagInfos[i].Name].GetHasNegativeFlag() {
+				if allFlags[*flagCollection.FlagInfos[i].Name].GetHasNegativeFlag() {
 					availableStartupFlags = append(availableStartupFlags, "no"+*flagCollection.FlagInfos[i].Name)
 				}
 			}
 		}
 	}
 
-	return flags, nil
+	argsWithoutStartupFlags := []string{os.Args[0]}
+
+	for _, arg := range os.Args[1:] {
+		isStartup := false
+		for _, availableStartupFlag := range availableStartupFlags {
+			if arg == "--"+availableStartupFlag || strings.Contains(arg, "--"+availableStartupFlag+"=") {
+				isStartup = true
+				break
+			}
+		}
+
+		if isStartup {
+			startupFlags = append(startupFlags, arg)
+		} else {
+			argsWithoutStartupFlags = append(argsWithoutStartupFlags, arg)
+		}
+	}
+
+	os.Args = argsWithoutStartupFlags
+
+	return allFlags, nil
```

Comment:
If you return 3 arguments, you don't need this hack with `os.Args` from this hidden place, instead, move the assignment to the `main` function, which is the right place for controversial things to leave.
```suggestion
	return allFlags, argsWithoutStartupFlags, nil
```

Reply:
Did some refactoring to make this cleaner